### DISCUSSION TITLE
Clean up option

### DISCUSSION THREAD

**Discussion by sfederici** [ScratchAddons/ScratchAddons#2072] [Discussion]
A few months ago I had noticed that the excellent Griffpatch's addon had replaced the "clean up" right-click option with a new "clean up (+)" that brought back the -at least to me- useless Scratch 2 clean up. But then I had noticed that in the menu I had both the Scratch 1.4/3 clean up (that I like, as it allows me to quickly browse the scripts of a project that I don't know and I see for the very first time) and the Scratch 2 clean up (that arranges the scripts in a more compact area, but that makes browsing the scripts of a project that you don't know very difficult, as in have to go down, the up and right, then down again etc etc). 

Would it be possible to add an option in the addon to choose your favorite clean up or -maybe better and easier- have both clean up in the contextual menu?

**Comment by cobaltt7** [comment]
So you want "Clean Up Blocks +" AND "Clean Up Blocks" in the right-click menu?
________________________________
From: sfederici ***@***.***>
Sent: Sunday, April 4, 2021 7:27:30 AM
To: ScratchAddons/ScratchAddons ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [ScratchAddons/ScratchAddons] Clean up option (#2072)


A few months ago I had noticed that the excellent Griffpatch's addon had replaced the "clean up" right-click option with a new "clean up (+)" that brought back the -at least to me- useless Scratch 2 clean up. But then I had noticed that in the menu I had both the Scratch 1.4/3 clean up (that I like, as it allows me to quickly browse the scripts of a project that I don't know and I see for the very first time) and the Scratch 2 clean up (that arranges the scripts in a more compact area, but that makes browsing the scripts of a project that you don't know very difficult, as in have to go down, the up and right, then down again etc etc).

Would it be possible to add an option in the addon to choose your favorite clean up or -maybe better and easier- have both clean up in the contextual menu?

**Reply by sfederici** [reply]
Yes. That one or, otherwise, a switch in Griffpatch's Addon to choose which kind of clean up you want (simple, "plus" or both).

**Comment by cobaltt7** [comment]
If it doesn't remove any features anymore, this could be enabled by default!
________________________________

**Reply by WorldLanguages** [reply]
@sfederici v1.13.0 update: you can now toggle `ENHANCE "CLEAN UP BLOCKS"` in the Scratch Addons settings page. You cannot, however, have both options in the menu

**Reply by sfederici** [reply]
Excellent! Even just having the option that one uses more is great. Thanks @RedGuy12 and @WorldLanguages!

**Reply by sfederici** [reply]
It works exactly as expected. Thanks

**Comment by sfederici** [comment]
I'm unable to show in the menu the standard "clean up blocks" without the "+". Enabling and Disabling the option in the "developer tools" addon doesn't make any difference (even reloading the project page)

**Reply by WorldLanguages** [reply]
Are you running the devtools extension or the addon? Keep in mind the extension will run even if the addon is enabled, and the extension will always enhance clean up blocks.

**Reply by sfederici** [reply]
Thanks, not it works again (even if I'm not clear why the setting was "lost")

**Reply by WorldLanguages** [reply]
It's simple, extension is preferred over addon, and extension will not respect addon settings. So if you don't want to use the predefined settings, you need to uninstall the extension and only use the addon
